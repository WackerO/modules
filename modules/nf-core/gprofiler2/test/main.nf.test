nextflow_process {

    name "Test Process GPROFILER2_GOST"
    script "../gost/main.nf"
    process "GPROFILER2_GOST"
    tag "modules"
    tag "modules_nfcore"
    tag "gprofiler2"
    tag "gprofiler2/gost"

    test("test_gprofiler2_gost") {
        config "./nextflow.config"
        when {
            params {
                outdir = "$outputDir"
                contrast_variable = "test"
            }
            process {
                """
                input[0] = [ id:'test', variable: 'c', reference:'r', target:'t' ]    // contrast info
                input[1] = [
                    [ id:'test', reference:'r', target:'t' ],           // meta map
                    file(params.test_data['mus_musculus']['genome']['deseq_results'], checkIfExists: true)
                ]
                input[2] = []                                           // no background
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
