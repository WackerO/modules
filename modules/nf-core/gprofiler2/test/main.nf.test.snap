{
    "test_gprofiler2_gost": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "all_enrich": [
                    
                ],
                "filtered_gmt": [
                    
                ],
                "plot_html": [
                    
                ],
                "plot_png": [
                    
                ],
                "rds": [
                    
                ],
                "session_info": [
                    
                ],
                "sub_enrich": [
                    
                ],
                "sub_plot": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "timestamp": "2023-11-07T09:16:38.826102148"
    }
}